PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -908,6 +920,65 @@ void AliAnalysisTaskEmcalEmbeddingHelper::UserCreateOutputObjects()
 {
   SetupEmbedding();
 
+  // Setup AliEventCuts
+  // NOTE: Need to define the base name here so that the property path is not ambiguous (due to otherwise only being `const char *`)
+  std::string baseName = "internalEventSelection";
+  bool res = fYAMLConfig.GetProperty({baseName, "enabled"}, fUseInternalEventSelection, false);
+  if (fUseInternalEventSelection)
+  {
+    AliDebugStream(1) << "Configuring AliEventCuts for internal event selection.\n";
+    // Handle manual cuts
+    res = fYAMLConfig.GetProperty({baseName, "useManualCuts"}, fUseManualInternalEventCuts, false);
+    if (res && fUseManualInternalEventCuts) {
+      fInternalEventCuts.SetManualMode();
+      // Implement these cuts by retrieving the event cuts object and setting them manually.
+    }
+
+    // Cuts which can be set regardless of manual cuts mode
+    // Centrality
+    std::vector <double> centralityRange;
+    bool res = fYAMLConfig.GetProperty({baseName, "centralityRange"}, centralityRange, false);
+    if (res) {
+      if (centralityRange.size() != 2) {
+        AliErrorStream() << "Passed centrality range with " << centralityRange.size() << " entries, but 2 values are required. Ignoring values.\n";
+      }
+      else {
+        AliDebugStream(1) << "Setting internal event centrality range to [" << centralityRange.at(0) << ", " << centralityRange.at(1) << ")\n";
+        fInternalEventCuts.SetCentralityRange(centralityRange.at(0), centralityRange.at(1));
+      }
+    }
+    // Z vertex
+    double zVertex = 999.0;
+    res = fYAMLConfig.GetProperty({baseName, "zVertex"}, zVertex, false);
+    if (res) {
+      AliDebugStream(1) << "Setting z vertex range of +/- " << zVertex << "\n";
+      fInternalEventCuts.SetMaxVertexZposition(zVertex);
+    }
+    // Trigger selection
+    bool useEventCutsAutomaticTriggerSelection = false;
+    res = fYAMLConfig.GetProperty({baseName, "useEventCutsAutomaticTriggerSelection"}, useEventCutsAutomaticTriggerSelection, false);
+    if (res && useEventCutsAutomaticTriggerSelection) {
+      // Use the autmoatic selection. Nothing to be done.
+      AliDebugStream(1) << "Using the automatic trigger selection from AliEventCuts.\n";
+    }
+    else {
+      // Use the cuts selected by SelectCollisionCandidates()
+      AliDebugStream(1) << "Using the trigger selection specified with SelectCollisionCandidates()\n.";
+      fInternalEventCuts.OverrideAutomaticTriggerSelection(fOfflineTriggerMask);
+    }
+
+    if (fCreateHisto) {
+      // This list will be owned by fOutput, so it won't be leaked.
+      TList * eventCutsOutput = new TList();
+      eventCutsOutput->SetOwner(kTRUE);
+      eventCutsOutput->SetName("EventCuts");
+
+      // Add the event cuts to the output
+      fInternalEventCuts.AddQAplotsToList(eventCutsOutput);
+      fOutput->Add(eventCutsOutput);
```

Comment:
fOutput is not initialized until a few lines later -- need to move it before this section

Reply:
Fixing now - I will push in a moment